rbac/management/group/view.py

```diff
@@ -761,8 +761,12 @@ def principals(self, request, uuid=None):
                         },
                     )
 
-                # Get the principal username option parameter and the limit and offset parameters too.
-                options[PRINCIPAL_USERNAME_KEY] = request.query_params.get(PRINCIPAL_USERNAME_KEY)
+                # Get the principal username option parameter and the limit and offset parameters too. Sometimes the UI
+                # sends "false" as the "principal username" query parameter. In that case we assume no filtering was
```

Comment:
well with this I am not sure and 😇 I'll wait for the arguments of someone more experienced because the
`principal_username` has this definition in open api spec "Parameter for filtering group principals by principal username using string contains search." ... so it seems to me that the fix should be done on the UI side = not to send request with this param

Reply:
Yeah without the full context on why this is happening, I'd agree with that. Given the spec, it should be a consumer of the API's responsibility to send a meaningful query param, otherwise I feel like we'd start coupling edge-cases to the logic?

Reply:
Yeah, I think you're both right... I thought about the fact that this might not be our responsibility to fix, but like I wanted to badly to fix this and try the stuff and make it work that I clearly made the wrong decision. I am going to talk to the UI team and let her now.

Thanks for the feedback!